src/Features/ERDDAP/hooks/TableDAPComponents.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * Render prop components to standardize the loading of datasets
+ */
+import * as React from "react"
+import { Alert } from "reactstrap"
+
+import { DataTimeSeries } from "Shared/timeSeries"
+import { PlatformTimeSeries } from "../types"
+
+import { useDataset, useDatasets } from "./tabledap"
+
+interface BaseProps {
+  /** Override default loading alert */
+  loading?: JSX.Element
+  /** Override default error alert */
+  error?: JSX.Element
+}
+
+interface UseDatasetsProps extends BaseProps {
+  children: (props: UseDatasetsRenderProps) => JSX.Element
```

Comment:
Codacy found an issue: ['JSX' is not defined.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)